File: src/main/java/model/SAC.java
package model;

public class SAC extends Amortizacao {

    @Override
    protected void atualizarValores(Emprestimo loan) {
        juros = saldo * loan.getTaxa();
        amortizacao = loan.getMontante() / loan.getTempo();
        pagamento = juros + amortizacao;
        saldo -= amortizacao;
    }
}

File: docs/sistemas/SAC.js
import AmortizationSystem from "./AmortizationSystem.js";

export default class SAC extends AmortizationSystem {
    constructor() {
        super("Sistema de Amortização Constante (SAC)");
    }
    atualizarValores(montante, tempo, taxa) {
        this.juros = this.saldo * taxa;
        this.amortizacao = montante / tempo;
        this.pagamento = this.juros + this.amortizacao;
        this.saldo -= this.amortizacao;
    }
}
